File: python/04_manage_partitions.py
import random, time
from pymilvus import MilvusClient, DataType

CLUSTER_ENDPOINT = "YOUR_CLUSTER_ENDPOINT"
TOKEN = "YOUR_CLUSTER_TOKEN"

# 1. Set up a Milvus client
client = MilvusClient(
    uri=CLUSTER_ENDPOINT,
    token=TOKEN 
)

# 2. Create a collection
client.create_collection(
    collection_name="quick_setup",
    dimension=5,
)

# 3. List partitions
res = client.list_partitions(collection_name="quick_setup")
print(res)

# Output
#
# ["_default"]



# 4. Create more partitions
client.create_partition(
    collection_name="quick_setup",
    partition_name="partitionA"
)

client.create_partition(
    collection_name="quick_setup",
    partition_name="partitionB"
)

res = client.list_partitions(collection_name="quick_setup")
print(res)

# Output
#
# ["_default", "partitionA", "partitionB"]



# 5. Check whether a partition exists
res = client.has_partition(
    collection_name="quick_setup",
    partition_name="partitionA"
)
print(res)

# Output
#
# True



res = client.has_partition(
    collection_name="quick_setup",
    partition_name="partitionC"
)
print(res)

# Output
#
# False



# 6. Load a partition independantly
client.release_collection(collection_name="quick_setup")

res = client.get_load_state(collection_name="quick_setup")
print(res)

# Output
#
# {
#     "state": "<LoadState: NotLoad>"
# }



client.load_partitions(
    collection_name="quick_setup",
    partition_names=["partitionA"]
)

res = client.get_load_state(collection_name="quick_setup")
print(res)

# Output
#
# {
#     "state": "<LoadState: Loaded>"
# }



res = client.get_load_state(collection_name="quick_setup", partition_name="partitionA")
print(res)

# Output
#
# {
#     "state": "<LoadState: Loaded>"
# }



res = client.get_load_state(collection_name="quick_setup", partition_name="partitionB")
print(res)

# Output
#
# {
#     "state": "<LoadState: NotLoad>"
# }



# 7. Release a partition
client.release_partitions(
    collection_name="quick_setup",
    partition_names=["partitionA"]
)

res = client.get_load_state(collection_name="quick_setup", partition_name="partitionA")
print(res)

# Output
#
# {
#     "state": "<LoadState: NotLoad>"
# }



# 8. Drop a partition
client.drop_partition(
    collection_name="quick_setup",
    partition_name="partitionB"
)

res = client.list_partitions(collection_name="quick_setup")
print(res)

# Output
#
# ["_default", "partitionA"]



# 9. Drop a collection
client.drop_collection(collection_name="quick_setup")

File: node/04_manage_partitions.js
const fs = require("fs")
const { MilvusClient, DataType, sleep } = require("@zilliz/milvus2-sdk-node")

const address = "YOUR_CLUSTER_ENDPOINT"
const token = "YOUR_CLUSTER_TOKEN"

async function main() {
    // 1. Set up a Milvus Client
    client = new MilvusClient({address, token});

    // 2. Create a collection in quick setup mode
    await client.createCollection({
        collection_name: "quick_setup",
        dimension: 5,
    });  

    // 3. List partitions
    res = await client.listPartitions({
        collection_name: "quick_setup"
    })

    console.log(res.partition_names)

    // Output
    // 
    // [ '_default' ]
    // 

    // 4. Create more partitions
    await client.createPartition({
        collection_name: "quick_setup",
        partition_name: "partitionA"
    })

    await client.createPartition({
        collection_name: "quick_setup",
        partition_name: "partitionB"
    })

    res = await client.listPartitions({
        collection_name: "quick_setup"
    })

    console.log(res.partition_names)

    // Output
    // 
    // [ '_default', 'partitionA', 'partitionB' ]
    // 

    // 5. Check whether a partition exists
    res = await client.hasPartition({
        collection_name: "quick_setup",
        partition_name: "partitionA"
    })

    console.log(res.value)

    // Output
    // 
    // true
    // 

    res = await client.hasPartition({
        collection_name: "quick_setup",
        partition_name: "partitionC"
    })
    
    console.log(res.value)

    // Output
    // 
    // false
    // 

    // 6. Load a partition indenpendantly
    await client.releaseCollection({
        collection_name: "quick_setup"
    })

    res = await client.getLoadState({
        collection_name: "quick_setup"
    })

    console.log(res.state)

    // Output
    // 
    // LoadStateNotLoad
    // 

    await client.loadPartitions({
        collection_name: "quick_setup",
        partition_names: ["partitionA"]
    })

    await sleep(3000)
    
    res = await client.getLoadState({
        collection_name: "quick_setup"
    })

    console.log(res.state)

    // Output
    // 
    // LoadStateLoaded
    // 

    res = await client.getLoadState({
        collection_name: "quick_setup",
        partition_name: "partitionA"
    })

    console.log(res.state)

    // Output
    // 
    // LoadStateLoaded
    // 

    res = await client.getLoadState({
        collection_name: "quick_setup",
        partition_name: "partitionB"
    })
    
    console.log(res.state)

    // Output
    // 
    // LoadStateLoaded
    // 

    // 7. Release a partition
    await client.releasePartitions({
        collection_name: "quick_setup",
        partition_names: ["partitionA"]
    })

    res = await client.getLoadState({
        collection_name: "quick_setup"
    })

    console.log(res.state)

    // Output
    // 
    // LoadStateNotLoad
    // 

    // 8 Drop a partition
    await client.dropPartition({
        collection_name: "quick_setup",
        partition_name: "partitionB"
    })

    res = await client.listPartitions({
        collection_name: "quick_setup"
    })

    console.log(res.partition_names)

    // Output
    // 
    // [ '_default', 'partitionA' ]
    // 

    // 9. Drop a collection
    await client.dropCollection({
        collection_name: "quick_setup"
    })
}

main()